packages/flutter/lib/src/rendering/sliver_persistent_header.dart

```diff
@@ -585,6 +763,80 @@ abstract class RenderSliverFloatingPersistentHeader extends RenderSliverPersiste
     _lastActualScrollOffset = constraints.scrollOffset;
   }
 
+  @override
+  void showOnScreen({
+    RenderObject descendant,
+    Rect rect,
+    Duration duration = Duration.zero,
+    Curve curve = Curves.ease,
+  }) {
+    if (showOnScreenConfiguration == null)
+      return super.showOnScreen(descendant: descendant, rect: rect, duration: duration, curve: curve);
+
+    assert(child != null || descendant == null);
+    // We prefer the child's coordinate space (instead of the sliver's) because
+    // it's easier for us to convert the target rect into target extents: when
+    // the sliver is sitting above the leading edge (not possible with pinned
+    // headers), the leading edge of the sliver and the leading edge of the child
+    // will not be aligned. The only exception is when child is null (and thus
+    // descendant == null).
+    final Rect childBounds = descendant != null
+      ? MatrixUtils.transformRect(descendant.getTransformTo(child), rect ?? descendant.paintBounds)
+      : rect;
+
+    // A stretch header can have a bigger childExtent than maxExtent.
+    final double effectiveMaxExtent = math.max(childExtent, maxExtent);
+    double minTargetExtent;
+    Rect targetRect;
+    switch (applyGrowthDirectionToAxisDirection(constraints.axisDirection, constraints.growthDirection)) {
+      case AxisDirection.up:
+        minTargetExtent = childExtent - (childBounds?.top ?? 0);
+        targetRect = showOnScreenConfiguration.ignoreLeading
+          ? _trim(childBounds, bottom: childExtent) : childBounds;
```

Comment:
nit: format so that the `: childBounds` is on a separate line aligned with the `?`

same below